### PR TITLE
No data bug performance

### PR DESCRIPTION
**Pull request by kheinz04**
Fixed bug for "No data found" not being properly displayed on the performance tab
image

Updated directive to more efficiently add and search for "noDataMsg" elements.

Updated formatting for test tab to display data better
- add "No data found" under header in addition to showing if no test fields are populated
- Removed some of the left padding to match alignment
- Updated scroll to only appear if there are enough items to make scroll necessary
Before: 
image
After:
image

Fixed div mismatch on feature widget

### CODE REVIEW

File: UI/src/components/widgets/build/directives/averageBuildDuration.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
     <div class="widget-heading">
         Average build duration
     </div>
-    <div hide-empty="['{{ buildView.buildDurationData }}']">
+    <div hide-empty="{{ buildView.buildDurationData.series }}">
```
Comment (nireeshT) [inline UI/src/components/widgets/build/directives/averageBuildDuration.html]: @kheinz04  check for undefined on buildView.buildDurationData

File: UI/src/components/widgets/build/directives/buildsPerDay.html [hunk 2]
```diff
@@ -2,7 +2,7 @@
     <div class="widget-heading">
         Builds per day
     </div>
-    <div hide-empty="['{{ buildView.lineData }}']">
+    <div hide-empty="['{{ buildView.lineData.series[0].data }}', '{{ buildView.lineData.series[1].data }}']">
```
Comment (nireeshT) [inline UI/src/components/widgets/build/directives/buildsPerDay.html]: @kheinz04 check for undefined or null or size

File: UI/src/components/widgets/performance/view.html [hunk 3]
```diff
@@ -1,198 +1,200 @@
...
-                                    </td>
-                                </tr>
-                                <tr>
-                                    <td ng-click="performanceView.showDetail(2)">
-                                        <span dash-status="alert"></span>
-                                        {{performanceView.bad.length}}
-                                    </td>
-                                </tr>
-                            </table>
+                                <table class="table violations" style="cursor:pointer;">
```
Comment (nireeshT) [inline UI/src/components/widgets/performance/view.html]: @kheinz04  can we get rid of table and move to div

File: UI/src/components/widgets/build/directives/averageBuildDuration.html (hunk 1 above)
Comment (kheinz04) [reply UI/src/components/widgets/build/directives/averageBuildDuration.html]: undefined assigned automatically through angular

File: UI/src/components/widgets/build/directives/buildsPerDay.html (hunk 2 above)
Comment (kheinz04) [reply UI/src/components/widgets/build/directives/buildsPerDay.html]: undefined assigned automatically through angular

File: UI/src/components/widgets/performance/view.html (hunk 3 above)
Comment (kheinz04) [reply UI/src/components/widgets/performance/view.html]: consider for future enhancement?

**Pull request merged**